openlcs/libs/common.py

```diff
@@ -287,6 +288,30 @@ def get_extension(file_name, sp_extensions):
     return os.path.splitext(file_name)
 
 
+def get_nvr_from_provide(provide):
+    """
+    Get nvr from parse the purl of the provide
+    """
+    purl = provide.get('purl')
+    purl = PackageURL.from_string(purl)
```

Comment:
Too many `purl.todict()` below, here could be `purl_dict = PackageURL.from_string(purl).to_dict()`

Reply:
Updated.